### PR TITLE
Load mod badge information from the FFZ API instead of assuming the URL.

### PR DESCRIPTION
**Pull request by pajlada**
This lets us load all sizes of the emote if they are available.

Channel with all versions of the mod badge: https://api.frankerfacez.com/v1/room/pajlada
Channel with only one version of the mod badge: https://api.frankerfacez.com/v1/room/apa420
Channel with no mod badge: https://api.frankerfacez.com/v1/room/forsen

A few "controversional" things I've done:
1) Image now has a custom "On successful network request" call, which the FFZ mod badge image uses to inject the green mod background. This is the `customOnSuccess` stuff in Image. As a result of this, `setPixmap` needed to be made public. I couldn't come up with a better solution
2) FFZ mod badge lives in the `FfzEmotes` file, so the name of that file isn't super logical anymore.
3) `FfzEmotes::loadChannel` needs to return information about the mod badge, so I added a new callback to it. I could've also just added it as a parameter to the first callback. I can easily do that upon request, just wasn't sure which felt cleaner (still not sure).

This PR is a response to #1291

### CODE REVIEW

**Comment by fourtf** [comment]
On first glance I dislike 1) since this should IMO be handled in a custom `MessageElement` similar to https://github.com/Chatterino/chatterino2/blob/master/src/messages/MessageElement.hpp#L203

**Comment by pajlada** [comment]
@fourtf Are you thinking a custom `MessageElement` type and a custom `ImageLayoutElement` that draws the green mod badge under it?
e.g. `ModBadgeElement` which in addToContainer adds a `ImageWithBackgroundLayoutElement` element which in turn has a paint element that not only draws its own "image pixmap, but also in this case the green mod badge background?

**Comment by fourtf** [comment]
@pajlada, yes

**Comment by fourtf** [comment]
@pajlada I think that reflects the "real world" situation where usually an emotes or badge is just an image but in this instance it's an image and a background. For the OOP structure it would make even more sense to make them extend EmoteElement and ImageLayoutElement

**Comment by pajlada** [comment]
@fourtf cool, will do

File: src/providers/ffz/FfzEmotes.cpp [hunk 1]
```diff
@@ -79,8 +79,37 @@ namespace {
 
         return {Success, std::move(emotes)};
     }
-    std::pair<Outcome, EmoteMap> parseChannelEmotes(const QJsonObject &jsonRoot)
+    std::tuple<Outcome, EmoteMap, boost::optional<EmotePtr>>
+        parseChannelResponse(const QJsonObject &jsonRoot)
```
Comment (fourtf) [inline src/providers/ffz/FfzEmotes.cpp]: Making a `parseChannelEmotes` and a `parseModBadge` function separately would work just as well. Here, The result is a tuple so need to either have comments or read the code to understand what the tuple items are.

Comment (fourtf) [reply src/providers/ffz/FfzEmotes.cpp]: Also, unrelated to this PR, the Outcome in the tuple could be removed it is unused.

**Comment by fourtf** [comment]
looks good to me

**Pull request merged**